llvm/lib/Analysis/ConstantFolding.cpp

```diff
@@ -4238,6 +4238,13 @@ static Constant *ConstantFoldScalableVectorCall(
 
     return ConstantInt::getFalse(SVTy);
   }
+  case Intrinsic::get_active_lane_mask: {
```

Comment:
I think we probably want to do this for fixed-width as well.

Reply:
As it stands, ConstantFolding can already handle this for fixed-width. Scalable seems to have been left behind: https://godbolt.org/z/snd7M5oer

Reply:
I see yeah, and as it stands the code in `ConstantFoldFixedVectorCall` is broken because it should return poison if `Op1` is zero. Once the LangRef is fixed we can always revisit this.